src/iuse_actor.cpp

```diff
@@ -277,14 +278,19 @@ std::optional<int> iuse_transform::use( Character *p, item &it, const tripoint &
     return result;
 }
 
-void iuse_transform::do_transform( Character *p, item &it ) const
+void iuse_transform::do_transform( Character *p, item &it, std::string variant_type ) const
 {
     item obj_copy( it );
     item *obj;
     // defined here to allow making a new item assigned to the pointer
     item obj_it;
     if( container.is_empty() ) {
         obj = &it.convert( target );
+        if( variant_type == "<any>" ) {
+            obj->set_itype_variant( obj->pick_random_variant() );
```

Comment:
This gives a debugmsg if they item doesn't have variants, but you don't check for that.